### PR TITLE
Coco attestation UI - Audit

### PR DESCRIPTION
**Pull request by mackdk**
## What does this PR change?

This PR adds a UI page to display the attestation reports on all systems accessible by the user under Audt > Confidential Computing. This listing behaves the same as the the one from https://github.com/uyuni-project/uyuni/pull/8597, without the system filtering, and reuses the same components.

Note: this change is merging to branch `coco-attestation-part2`.

## GUI diff

After:
image

- [X] **DONE**

## Documentation
- Documentation WIP https://github.com/SUSE/spacewalk/issues/23652

- [ ] **DONE**

## Test coverage
- Unit tests were added

- [X] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/23563

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by Etheryte — APPROVED** [review]

File: java/code/src/com/suse/manager/attestation/test/AttestationManagerTest.java [hunk 1]
```diff
@@ -136,34 +137,104 @@ public void testCreateAttestationAction() throws TaskomaticApiException {
...
+        createFakeAttestationReport(user, server);
+        createFakeAttestationReport(user, server3);
 
-        ServerCoCoAttestationReport report2 = mgr.initializeReport(user, server);
-        mgr.initializeResults(user, report2);
-        fakeSuccessfullAttestation(report2);
         HibernateFactory.getSession().flush();
         HibernateFactory.commitTransaction();
         HibernateFactory.getSession().clear();
+        commitHappened();
```
Comment (mcalmer) [inline java/code/src/com/suse/manager/attestation/test/AttestationManagerTest.java:151]: Have you checked, that this really cleanup all the config, reports and results?

File: java/code/src/com/suse/manager/model/attestation/AttestationFactory.java [hunk 2]
```diff
@@ -239,4 +255,34 @@ public List<ServerCoCoAttestationReport> listCoCoAttestationReports(Server serve
...
                 .list();
     }
+
+    /**
+     * Return a list of reports for a given server with filters
+     * @param user the user
+     * @param pc page control object
+     * @return returns a list or reports
+     */
+    public List<ServerCoCoAttestationReport> listCoCoAttestationReports(User user, PageControl pc) {
```
Comment (mcalmer) [inline java/code/src/com/suse/manager/model/attestation/AttestationFactory.java:265]: I thought we have this method already. But this one is listing for a User, while the other is listing for a Server.
Maybe we should encode this in the names?
listCoCoAttestationReportsByUser() and listCoCoAttestationReportsByServer() ?

File: java/code/src/com/suse/manager/attestation/test/AttestationManagerTest.java (hunk 1 above)
Comment (mackdk) [reply java/code/src/com/suse/manager/attestation/test/AttestationManagerTest.java:151]: Yes, toggling the `committed` flag will trigger the removal of the organization at the end of the process in [JMockBaseTestCaseWithUser](https://github.com/uyuni-project/uyuni/blob/98f5f0790457610509e637c8e5638e5e039063f8/java/code/src/com/redhat/rhn/testing/JMockBaseTestCaseWithUser.java#L63C1-L68C10):

```java
        // If at some point we created a user and committed the transaction, we need
        // clean up our mess
        if (committed) {
            OrgFactory.deleteOrg(user.getOrg().getId(), user);
            commitAndCloseSession();
        }
```

File: java/code/src/com/suse/manager/model/attestation/AttestationFactory.java (hunk 2 above)
Comment (mackdk) [reply java/code/src/com/suse/manager/model/attestation/AttestationFactory.java:265]: Yes, good point. I will change the naming :+1:

**Review by lucidd — APPROVED** [review]

**Pull request merged**